spec.html

```diff
@@ -49144,7 +49140,7 @@ <h1>%Symbol.toStringTag%</h1>
 
 <emu-clause id="sec-memory-model">
   <h1>Memory Model</h1>
-  <p>The memory consistency model, or <dfn>memory model</dfn>, specifies the possible orderings of Shared Data Block events, arising via accessing TypedArray instances backed by a SharedArrayBuffer and via methods on the Atomics object. When the program has no data races (defined below), the ordering of events appears as sequentially consistent, i.e., as an interleaving of actions from each agent. When the program has data races, shared memory operations may appear sequentially inconsistent. For example, programs may exhibit causality-violating behaviour and other astonishments. These astonishments arise from compiler transforms and the design of CPUs (e.g., out-of-order execution and speculation). The memory model defines both the precise conditions under which a program exhibits sequentially consistent behaviour as well as the possible values read from data races. To wit, there is no undefined behaviour.</p>
+  <p>The memory consistency model, or <dfn>memory model</dfn>, specifies the possible orderings of Shared Data Block events, arising via accessing TypedArray instances backed by a SharedArrayBuffer and via methods on the Atomics object. When the program has no data races (defined below), the ordering of events appears as sequentially consistent, i.e., as an interleaving of actions from each agent. When the program has data races, shared memory operations may appear sequentially inconsistent. For example, programs may exhibit causality violation and other astonishing behaviour. These astonishing behaviours arise from compiler transforms and the design of CPUs (e.g., out-of-order execution and speculation). The memory model defines both the precise conditions under which a program exhibits sequentially consistent behaviour as well as the possible values read from data races. To wit, there is no undefined behaviour.</p>
```

Comment:
I don't like any of these changes. "causality-violating behaviour" is fine. "astonishments" is a perfectly reasonable word.

Reply:
I disagree about "astonishments". Astonishment is the feeling that something astonishing causes, not the thing itself. From Merriam-Webster:

> a feeling of great surprise and wonder : the state of being astonished

The change is definitely an improvement. The "causality violation" change is fine either way but I think reads better the way I have it after the change to "astonishing behaviour".

Reply:
Words can have more than one definition.

<img width="763" alt="Screenshot 2024-07-25 at 6 48 23 PM" src="https://github.com/user-attachments/assets/b9c16ff0-1fc8-4306-8f1a-ddda94bff889">

Reply:
That is a far less common usage, possibly even archaic. I'm a native speaker and don't think I've ever heard that usage in my life. I stand by my assertion that the change is an improvement, even if the previous version was not incorrect.

Reply:
It's OK to use slightly more obscure words, especially when the meaning is obvious. The text does not need to be maximally dry.